#!/usr/bin/env python
PACKAGE = "lidar"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("tilt_speed", int_t , 0, "Inditates the speed at which the lidar should tilt. If '0' the speed is computed automagically to ensure continuous movement.", 0, 0, 100)
gen.add("down_tilt_angle", int_t, 0, "Maximum downwards pitch angle in degrees, 90 is horizontal, higher angle point down.", 180, 0, 180)
gen.add("up_tilt_angle", int_t, 0, "Maximum upwards pitch angle in degrees, 90 is horizontal, higher angle point down.", 90, 0, 180)
gen.add("tilt_enable",   bool_t,   0, "Enable the tilting motion",  True)

exit(gen.generate(PACKAGE, "lidar", "LidarTilt"))